src/core/data/ndarray/ndarray_vector.h

```diff
@@ -64,10 +64,71 @@ struct NdArrayViewer
 };
 
 
+
+template<typename Array, typename Mask>
+class MaskedView
+{
+public:
+    static auto constexpr dimension = Array::dimension;
```

Comment:
we should probably static_assert mask and array have compatible dimensions ?

Reply:
I'm guessing there'd be a compile error, can confirm

Reply:
MaskedView takes a mask as template, so the dim is implicit